runtime/src/accounts_db.rs

```diff
@@ -3179,31 +3316,35 @@ impl AccountsDb {
         self.combine_ancient_slots(old_slots);
     }
 
-    fn create_ancient_append_vec(&self, slot: Slot) -> Option<(Slot, Arc<AccountStorageEntry>)> {
-        let (new_ancient_storage, _time) =
+    fn create_ancient_append_vec(
+        &self,
+        slot: Slot,
+    ) -> (Option<(Slot, Arc<AccountStorageEntry>)>, u64) {
```

Comment:
Any chance this function (and all that use it) could return a `Duration` instead of a `u64`?

To fully fix this, I think that `get_store_for_shrink()` would also need to be updated to return a `Duration` as well. That can be done in a separate PR. I think the simplest impl here would be to add a `Duration::from_micros()` around the `time` param after calling `get_store_for_shrink()`. What do you think?

https://github.com/solana-labs/solana/blob/3ed2e0ce26154588eb0585d917cf77ac645ab39d/runtime/src/accounts_db.rs#L3124-L3129